localstack/services/kms/models.py

```diff
@@ -0,0 +1,532 @@
+    # "AWS KMS generates the data key. Then it encrypts a copy of the data key under a symmetric encryption KMS key that
+    # you specify."
+    #
+    # Data keys are symmetric, data key pairs are asymmetric.
+    def _populate_metadata(
+        self, create_key_request: CreateKeyRequest, account_id: str, region: str
+    ) -> None:
+        self.metadata = KeyMetadata()
+        # Metadata fields coming from a creation request
+        #
+        # Please keep in ind that tags of a key could be present in the request, they are not a part of metadata. At
+        # least in the sense of DescribeKey not returning them with the rest of the metadata. Instead, tags are more
+        # like aliases:
+        # https://docs.aws.amazon.com/kms/latest/APIReference/API_DescribeKey.html
+        # "DescribeKey does not return the following information: ... Tags on the KMS key."
+        self.metadata["Description"] = create_key_request.get("Description") or ""
+        self.metadata["KeyUsage"] = create_key_request.get("KeyUsage") or "ENCRYPT_DECRYPT"
+        self.metadata["MultiRegion"] = create_key_request.get("MultiRegion") or False
+        self.metadata["Origin"] = create_key_request.get("Origin") or "AWS_KMS"
+        # https://docs.aws.amazon.com/kms/latest/APIReference/API_CreateKey.html#KMS-CreateKey-request-CustomerMasterKeySpec
+        # CustomerMasterKeySpec has been deprecated, still used for compatibility. Is replaced by KeySpec.
+        # The meaning is the same, just the name differs.
+        self.metadata["KeySpec"] = (
+            create_key_request.get("KeySpec")
+            or create_key_request.get("CustomerMasterKeySpec")
+            or "SYMMETRIC_DEFAULT"
+        )
+        self.metadata["CustomerMasterKeySpec"] = self.metadata.get("KeySpec")
+
+        # Metadata fields AWS introduces automatically
+        self.metadata["AWSAccountId"] = account_id or get_aws_account_id()
+        self.metadata["CreationDate"] = time.time()
+        self.metadata["Enabled"] = True
+        self.metadata["KeyManager"] = "CUSTOMER"
+        self.metadata["KeyState"] = "Enabled"
+        # https://docs.aws.amazon.com/kms/latest/developerguide/multi-region-keys-overview.html
+        # "Notice that multi-Region keys have a distinctive key ID that begins with mrk-. You can use the mrk- prefix to
+        # identify MRKs programmatically."
+        key_id_prefix = ""
+        if self.metadata.get("MultiRegion"):
+            key_id_prefix = "mrk-"
+        self.metadata["KeyId"] = key_id_prefix + str(uuid.uuid4())
+        self.metadata["Arn"] = kms_key_arn(self.metadata.get("KeyId"), account_id, region)
+
+        self._populate_encryption_algorithms(
+            self.metadata.get("KeyUsage"), self.metadata.get("KeySpec")
+        )
+        self._populate_signing_algorithms(
+            self.metadata.get("KeyUsage"), self.metadata.get("KeySpec")
+        )
+
+    def add_tags(self, tags: List) -> None:
+        try:
```

Comment:
Better avoid logic that relies on `AttributeError` - class instance attributes should be well-defined, and always initialized with a value in the initializer.

Let's add `tags: Dict[str, str]` to the definition of attributes of this class (at the top of the class definition), and initialize `self.tags = {}` in `__init__(..)`.

Reply:
Done.